### PR TITLE
Automatically Install Modules If Not Found

### PR DESCRIPTION
**Pull request by TechStudent11**
# Related Issue/Addition to code
Please delete options that are not relevant.

- Automatically Install Modules If Not Found

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Proposed Changes
- Automatically Install Modules If Not Found

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by Kurdakov** [comment]
see https://www.pythonpool.com/pip-vs-pip3/ your assumption
if platform.system() == "Windows":
    pip_exe = "pip"
  else:
    pip_exe = "pip3"

that in Windows it's pip is incorrect.
pip3 though should always do the job, but then again, it does not look as a good practice - almost nobody uses automatic installation.

File: YouTubeSpammerPurge.py [hunk 1]
```diff
@@ -70,22 +70,34 @@
...
+  from google.auth.transport.requests import Request
+except ModuleNotFoundError as e:
+  print("Modules Not Found. Attempting to install...")
+  pip_exe = None
+  if platform.system() == "Windows":
+    pip_exe = "pip"
+  else:
+    pip_exe = "pip3"
 
+  os.system(f"{pip_exe} install -r requirements.txt")
```
Comment (compsup) [inline YouTubeSpammerPurge.py:97]: Use `subprocess.check_call([sys.executable, "-m", "pip", "install", "-r", "requirements.txt"])`

**Review by compsup — CHANGES_REQUESTED** [review]
Use a try statement to attempt to install using pip3 first and if that fails try pip:

```
try:
 subprocess.call(['pip3', 'install', '-r requirements.txt'])
except:
 subprocess.call(['pip', 'install', '-r requirements.txt'])
```
I realize the issue if the OS has both pip and pip3 where pip is for python2 and pip3 is for python3, trying to find solutions

**Comment by Kurdakov** [comment]
> Use a try statement to attempt to install using pip3 first and if that fails try pip:
> 
> ```
> try:
>   subprocess.check_call(['pip3', 'install', '-r requirements.txt'])
> except:
>   try:
>     subprocess.check_call(['pip', 'install', '-r requirements.txt'])
>   except:
>     pass
> ```

pip3 always works if python version 3 is installed (pip is just symlink to pip3 in this case), so in suggested snippet instead modules could be erroneously installed to python2 (if it is present and is default installation)

**Comment by compsup** [comment]
> pip3 always works if python version 3 is installed (pip is just symlink to pip3 in this case), so in suggested snippet instead modules could be erroneously installed to python2 (if it is present and is default installation)

Except in linux installing using pip will only install python2 modules, unless im miss-reading what you said

**Comment by TechStudent11** [comment]
> > pip3 always works if python version 3 is installed (pip is just symlink to pip3 in this case), so in suggested snippet instead modules could be erroneously installed to python2 (if it is present and is default installation)
> 
> Yes thats correct, so only need to try pip3

got it.

**Review by compsup — APPROVED** [review]
Looks good

**Comment by TechStudent11** [comment]
Closing in favor of https://github.com/ThioJoe/YT-Spammer-Purge/discussions/213#discussioncomment-1949942

**Comment by TechStudent11** [comment]
Reopening just in case that get's merged.

**Comment by ThioJoe** [comment]
I'd rather not have things auto install, especially because it could really mess up someone's virtual environments if they forget to load it

**Comment by TechStudent11** [comment]
ok got it

**Pull request closed without merging**